engine/src/ast/function_expr.rs

```diff
@@ -76,6 +76,10 @@ impl<'s> FunctionCallExpr<'s> {
                 FunctionCallArgExpr::Literal(literal) => literal.into(),
             })
         }
+        for i in (values.len() - self.function.args.len())..self.function.opt_args.len() {
+            let opt_arg = &self.function.opt_args[i];
```

Comment:
Nit: use proper iterator over `self.function.opt_args` when possible - it usually both reads and works better (because iterators guarantee that index can't be out of bounds etc.).

Reply:
In this case you might want to do something like

```rust
for opt_arg in &self.function.opt_args[values.len() - self.function.args.len()..] {
 ..
}
```

Although that brings to the next question - is it somehow guaranteed that `values.len() >= self.function.args.len()` and won't overflow?

Reply:
It should be guaranteed because in the parsing step, we do not parse less than `self.function.args.len()` and not more then `self.function.args.len() + self.function.opt_args.len()`